### ISSUE TITLE
ASP.NET 2.0 doesn't work

### ISSUE THREAD

**Issue by rspaulino** [Azure/draft#484]
Hi,

I'm a lil confuse because when I see the changelog of v0.9.0 it mention "bumps the ASP.NET pack to 2.0" so I'm assuming that draft would work with asp.net core 2.0 but when I ran a simple new projects.

`dotnet new mvc`

and then ran 

`draft create --debug`

I get the following result:

```
Error: <nil> 
DEBU[0000] JSON:	90.361723 ()                           
DEBU[0000] SVG:	6.544119 ()                             
DEBU[0000] C#:	2.343193 (#178600)                       
DEBU[0000] XML:	0.613372 ()                             
DEBU[0000] CSS:	0.109026 (#563d7c)                      
DEBU[0000] JavaScript:	0.028566 (#f1e05a)               
--> Draft detected the primary language as JSON with 90.361723% certainty.
Error: Could not find a pack Q_Q
```

is there any flag I can pass to force the language detection to just take my word that it's asp.net? I try playing around with the linguist's .gitattributes support but didn't have much luck, even try *.csproj linguist-language=csharp so this in one of the issues.

but anyways if anyone could help me on how to get this going, right now I run the command manually to compile and deploy but was looking to simplify the process like it show in the tutorial videos.

Thanks in advance.

**Comment by bacongobbler** [comment]
Hey! Gitattributes support was only recently merged so you’ll have to try one of the canary builds in order for that to work. :)

There’s a similar ticket via https://github.com/Azure/draft/issues/476 that poses an interesting solution. How does that idea sound to you as a feature?

**Comment by rspaulino** [comment]
@bacongobbler Thanks for you rapid response.

Is asp.net 2.0 supported in draft v0.9.0? and just so I'm clear, in the latest release:

> v0.9.0 adds a Swift pack, bumps the ASP.NET pack to 2.0 and added .gitattributes support to draft create to override the language detection handled with github linguist.

Do you mean this particular attribute is in canary "*.csproj linguist-language=csharp" or all .gitattributes were not release with v0.9.0

as for #476 the only issue I can see if that for example in .net you can have SPA project which have .js & .cs technically .js is more popular but creating chart for .js from actually run the project correctly.

**Comment by jstrachan** [comment]
I wonder if we need a weighting on languages too; e.g. the presence of a "pom.xml" for Java should outweigh however much `.js` and `.java` - I wonder if a `.csproj` could be weighted higher than `.js` and `.cs` in the same way?

i.e. treat language specific build configuration tools (maven / gradle / .net / npm et al) as higher priority to general purpose language files?

**Comment by bacongobbler** [comment]
> I wonder if we need a weighting on languages too

That would be a good feature request at https://github.com/github/linguist to weigh certain file extensions higher than .js and .cs. Right now linguist trains the bayesian classifier against example projects. It might also be [good to supply feature weights](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.377.2584&rep=rep1&type=pdf) to the [naive bayesian classifier](https://github.com/jbrukh/bayesian). If anyone's interested in getting involved with naive bayesian classifiers, that would be a great starter project.

**Comment by rspaulino** [comment]
@jstrachan I agree with you. In ASP.NET or .NET we use solutions (*.sln) or projects(C# *.csproj), the only time I see .cs individual file is for script purposes.

I find a way around this and it's using --pack

`draft create --pack=csharp`

In my case draft was looking for the pack in /usr/local/etc/draft/packs/

so I have to go to /usr/local/etc/draft/packs/github.com/Azure/draft/packs and move the csharp in the location above now it work perfect or at least more than the default, I haven't test it with complex apps but it should work.

**Comment by shahiddev** [comment]
Looks like we still have an issue with draft detecting the correct language when it comes to .net
```C:\Repos\drafttest
λ dotnet new mvc
The template "ASP.NET Core Web App (Model-View-Controller)" was created successfully.
This template contains technologies from parties other than Microsoft, see https://aka.ms/template-3pn for details.

Processing post-creation actions...
Running 'dotnet restore' on C:\Repos\drafttest\drafttest.csproj...
  Restoring packages for C:\Repos\drafttest\drafttest.csproj...
  Restore completed in 142.57 ms for C:\Repos\drafttest\drafttest.csproj.
  Generating MSBuild file C:\Repos\drafttest\obj\drafttest.csproj.nuget.g.props.
  Generating MSBuild file C:\Repos\drafttest\obj\drafttest.csproj.nuget.g.targets.
  Restore completed in 5.48 sec for C:\Repos\drafttest\drafttest.csproj.

Restore succeeded.


C:\Repos\drafttest
λ draft create
--> Draft detected Coq (45.110470%)
--> Could not find a pack for Coq. Trying to find the next likely language match...
--> Draft detected JSON (23.281315%)
--> Could not find a pack for JSON. Trying to find the next likely language match...
--> Draft detected JavaScript (17.572730%)
--> Ready to sail
```
